src/daft-core/src/utils/stats.rs

```diff
@@ -0,0 +1,82 @@
+use common_error::DaftResult;
+
+use crate::{
+    array::{
+        ops::{DaftCountAggable, DaftSumAggable, GroupIndices, VecIndices},
+        prelude::{Float64Array, UInt64Array},
+    },
+    count_mode::CountMode,
+};
+
+#[derive(Clone, Copy, Default, Debug)]
+pub struct Stats {
+    pub sum: f64,
+    pub count: u64,
```

Comment:
Is the u64 version of count used anywhere? If not it may make sense to just have it as an f64 here, then you don't have to do the `as f64` cast in any calculation loop.

Reply:
This is a great catch. Ya, without performing an early cast, I would perform a cast every single iteration. Not good...

Updated and pushed!